service/awsconfig/v10/resource/ebsvolume/delete.go

```diff
@@ -20,39 +20,38 @@ func (r *Resource) EnsureDeleted(ctx context.Context, obj interface{}) error {
 	// Get both the Etcd volume and any Persistent Volumes.
 	etcdVolume := true
 	persistentVolume := true
-
 	volumes, err := r.service.ListVolumes(customObject, etcdVolume, persistentVolume)
 	if err != nil {
 		return microerror.Mask(err)
 	}
 
 	if len(volumes) > 0 {
-		r.logger.LogCtx(ctx, "level", "info", "message", fmt.Sprintf("deleting %d EBS volumes", len(volumes)))
+		r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("deleting %d EBS volumes", len(volumes)))
 
 		// First detach any attached volumes without forcing but shutdown the
 		// instances.
-		force := false
-		shutdown := true
-
 		for _, vol := range volumes {
 			for _, a := range vol.Attachments {
-				err := r.service.DetachVolume(ctx, vol.VolumeID, a, force, shutdown)
+				force := false
+				shutdown := true
+				wait := false
```

Comment:
Putting the variables into the inner scopes. I am not worried about duplicated allocations in the memory stack. We do not have such problems. This here is safer though to prevent nasty side effects in the upper scopes.

Reply:
I don't think there would be an allocation in any case.